src/GlobalCtx.tsx

```diff
@@ -1,15 +1,69 @@
 import type { ApiPromise, WsProvider } from "@polkadot/api";
 import type { TypeRegistry } from "@polkadot/types";
+import { formatBalance } from "@polkadot/util";
 import { createContext, useContext } from "react";
 import { GlobalCtxProvider, type Status } from "./GlobalCtxProvider";
 
+export class TokenProperties {
+  tokenSymbol: string;
+  tokenDecimals: number;
+
+  constructor(args: { tokenSymbol: string; tokenDecimals: number }) {
+    this.tokenSymbol = args.tokenSymbol;
+    this.tokenDecimals = args.tokenDecimals;
+  }
+
+  // `default` is a reserved keyword
+  static preset(): TokenProperties {
+    return new TokenProperties({
+      tokenSymbol: "UNIT",
+      tokenDecimals: 12,
+    });
+  }
+
+  /** Query the Polkadot API for the chain properties, will throw if either are missing */
+  static async fromApi(api: ApiPromise): Promise<TokenProperties> {
+    const systemProperties = await api.rpc.system.properties();
+    return new TokenProperties({
+      // We have a single token, hence the [0]
+      tokenDecimals: systemProperties.tokenDecimals.unwrap()[0].toNumber(),
+      tokenSymbol: systemProperties.tokenSymbol.unwrap()[0].toString(),
+    });
+  }
+
+  /** Converts a given value in the chain's respective UNIT to Plancks (amount denoted by the `tokenDecimals` field). */
+  unitToPlanck(unit: number): number {
+    return unit * this.tokenDecimals;
+  }
+
+  planckToUnit(planck: number): number {
+    return planck / this.tokenDecimals;
+  }
+
+  formatUnit(unit: number | bigint): string {
+    // 0 != BigInt(0) hence the ||
+    if (unit === 0 || (typeof unit === "bigint" && BigInt(0) === unit)) {
```

Comment:
I think this does the same but shorter:
```suggestion
    if (unit === 0 || unit === 0n) {
```
The `n` behind the second 0 is indicating that it is a `bigint`

Reply:
oh nice